File: ios/ExpoRoomPlanModule.swift
import ExpoModulesCore
import UIKit

@available(iOS 17.0, *)
public class ExpoRoomPlanModule: Module {
    private var captureViewController: RoomPlanCaptureViewController?

    public func definition() -> ModuleDefinition {
        Name("ExpoRoomPlan")

        Events("onDismissEvent")

        AsyncFunction("startCapture") {
            (scanName: String, exportType: String, sendFileLoc: Bool) in
            DispatchQueue.main.async {
                let captureVC = RoomPlanCaptureViewController()

                captureVC.scanName = scanName
                captureVC.modalPresentationStyle = .fullScreen
                captureVC.exportType = exportType
                captureVC.sendFileLoc = sendFileLoc

                captureVC.onDismiss = { eventData in
                    self.sendEvent("onDismissEvent", eventData)
                }

                guard
                    let rootVC = UIApplication.shared.connectedScenes
                        .compactMap({ ($0 as? UIWindowScene)?.keyWindow })
                        .first?.rootViewController
                else {
                    return
                }

                rootVC.present(captureVC, animated: true, completion: nil)
                self.captureViewController = captureVC
            }
        }

        AsyncFunction("stopCapture") {
            DispatchQueue.main.async {
                self.captureViewController?.stopSession()
                self.captureViewController?.dismiss(
                    animated: true,
                    completion: nil
                )
            }
        }
    }
}

File: android/src/main/java/expo/modules/roomplan/ExpoRoomplanModule.kt
package expo.modules.roomplan

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import expo.modules.kotlin.Promise
import expo.modules.kotlin.events.EventEmitter

class ExpoRoomPlanModule : Module() {
  
  override fun definition() = ModuleDefinition {
    // Module name - should match your TypeScript module name
    Name("ExpoRoomPlan")

    // Async function to start capture
    AsyncFunction("startCapture") { scanName: String, exportType: String, sendFileLoc: Boolean, promise: Promise ->
      promise.reject("NOT_IMPLEMENTED", "RoomPlan SDK is not available on Android.")
    }

    // Async function to stop capture
    AsyncFunction("stopCapture") { promise: Promise ->
      promise.reject("NOT_IMPLEMENTED", "RoomPlan SDK is not available on Android.")
    }
  }
}